modules/taboolaBidAdapter.js

```diff
@@ -0,0 +1,255 @@
+'use strict';
+
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER} from '../src/mediaTypes.js';
+import {config} from '../src/config.js';
+import {getWindowSelf, getWindowTop} from '../src/utils.js'
+import {getStorageManager} from '../src/storageManager.js';
+
+const BIDDER_CODE = 'taboola';
+const GVLID = 42;
+const CURRENCY = 'USD';
+export const END_POINT_URL = 'https://taboolahb.bidder.taboolasyndication.com';
+const USER_ID = 'user-id';
+const STORAGE_KEY = `taboola global:${USER_ID}`;
+const COOKIE_KEY = 'trc_cookie_storage';
+
+/**
+ * try to extract User Id by that order:
+ *  local storage
+ *  first party cookie
+ *  rendered trc
+ *  new user set it to 0
+ */
+export const userData = {
+  storageManager: getStorageManager(GVLID, BIDDER_CODE),
+  getUserId: () => {
+    const {getFromLocalStorage, getFromCookie, getFromTRC} = userData;
+
+    try {
+      return getFromLocalStorage() || getFromCookie() || getFromTRC();
+    } catch (ex) {
+      return 0;
+    }
+  },
+  getFromCookie() {
+    const {cookiesAreEnabled, getCookie} = userData.storageManager;
+    if (cookiesAreEnabled()) {
+      const cookieData = getCookie(COOKIE_KEY);
+      const userId = userData.getCookieDataByKey(cookieData, USER_ID);
+      if (userId) {
+        return userId;
+      }
+    }
+  },
+  getCookieDataByKey(cookieData, key) {
+    const [, value = ''] = cookieData.split(`${key}=`)
+    return value;
+  },
+  getFromLocalStorage() {
+    const {hasLocalStorage, localStorageIsEnabled, getDataFromLocalStorage} = userData.storageManager;
+
+    if (hasLocalStorage() && localStorageIsEnabled()) {
+      return getDataFromLocalStorage(STORAGE_KEY);
+    }
+  },
+  getFromTRC() {
+    return window.TRC ? window.TRC.user_id : 0;
+  }
+}
+
+export const internal = {
+  getPageUrl: (refererInfo = {}) => {
+    if (refererInfo.canonicalUrl) {
+      return refererInfo.canonicalUrl;
+    }
+
+    if (config.getConfig('pageUrl')) {
+      return config.getConfig('pageUrl');
+    }
+
+    try {
+      return getWindowTop().location.href;
+    } catch (e) {
+      return getWindowSelf().location.href;
```

Comment:
no that i can think of ... this is the fallback if you can't reach to the winodw.top because it's not a friandly iframe ...
this code is used in other adapters so i think we are good here